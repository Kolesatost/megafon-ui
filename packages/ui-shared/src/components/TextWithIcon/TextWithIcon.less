@import '~@megafon/ui-core/styles/base.less';

@b: e('.mfui-text-with-icon');

@{b} {
    &__header {
        margin-bottom: 24px;
    }

    &_centering-on-mobile &__header {
        @media @mobileSM {
            text-align: center;
        }
    }
}
