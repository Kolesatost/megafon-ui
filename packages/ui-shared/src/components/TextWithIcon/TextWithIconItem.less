@import '~@megafon/ui-core/styles/base.less';

@b: e('.mfui-text-with-icon-item');

@{b} {
    display: flex;
    align-items: center;

    &:not(:first-of-type) {
        margin-top: 16px;

        @media @mobileSM {
            margin-top: 24px;
        }
    }

    &__svg-icon {
        width: 40px;
        min-width: 40px;
        height: 40px;
        min-height: 40px;

        margin-right: 16px;
        overflow: hidden;
    }

    &_centering-on-mobile {
        @media @mobileSM {
            flex-direction: column;
        }
    }

    &_centering-on-mobile &__text {
        @media @mobileSM {
            margin-top: 12px;

            text-align: center;
        }
    }

    &_centering-on-mobile &__svg-icon {
        @media @mobileSM {
            margin-right: 0;
        }
    }
}
