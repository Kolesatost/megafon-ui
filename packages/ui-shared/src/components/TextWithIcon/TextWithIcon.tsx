import * as React from 'react';
import { Header, Grid, GridColumn } from '@megafon/ui-core';
import { cnCreate, filterDataAttrs, IFilterDataAttrs } from '@megafon/ui-helpers';
import * as PropTypes from 'prop-types';
import './TextWithIcon.less';

export interface IItem {
    /** Текст */
    text: string;
    /** Иконка */
    icon: React.ReactNode;
}

export interface ITextWithIconProps extends IFilterDataAttrs {
    /** Список строк с иконками */
    items: IItem[];
    /** Заголовок */
    title?: string;
    /** Ссылка на корневой элемент */
    rootRef?: React.Ref<HTMLDivElement>;
    /** Дополнительный класс для корневого элемента */
    className?: string;
}

const cn = cnCreate('mfui-text-with-icon');
const TextWithIcon: React.FC<ITextWithIconProps> = ({ items, title, rootRef, dataAttrs, className }) => (
    <div className={cn([className])} ref={rootRef} {...filterDataAttrs(dataAttrs)}>
        <Grid>
            <GridColumn {...{ mobile: '12', tablet: '7', desktop: '6', wide: '6' }}>
                {title && (
                    <Header className={cn('header')} as="h5">
                        {title}
                    </Header>
                )}
                {items.map(({ text, icon }, i) => (
                    <div className={cn('item')} key={i}>
                        <div className={cn('svg-icon')}>{icon}</div>
                        <div className={cn('text')}>{text}</div>
                    </div>
                ))}
            </GridColumn>
        </Grid>
    </div>
);

TextWithIcon.propTypes = {
    items: PropTypes.arrayOf(
        PropTypes.shape({
            text: PropTypes.string.isRequired,
            icon: PropTypes.node.isRequired,
        }).isRequired,
    ).isRequired,
    title: PropTypes.string,
    rootRef: PropTypes.oneOfType([
        PropTypes.func,
        PropTypes.oneOfType([PropTypes.shape({ current: PropTypes.elementType }), PropTypes.any]),
    ]),
    dataAttrs: PropTypes.objectOf(PropTypes.string.isRequired),
    className: PropTypes.string,
};

export default TextWithIcon;
