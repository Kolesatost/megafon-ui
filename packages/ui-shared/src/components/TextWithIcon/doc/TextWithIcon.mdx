---
name: TextWithIcon
menu: Конструктор
route: components/text-with-icon
---

import Example from './TextWithIcon.example.mdx';
import Props from './TextWithIcon.props.mdx';
import TextWithIcon from '../TextWithIcon';

# TextWithIcon

<CopyToClipboardBox text="import { TextWithIcon } from '@megafon/ui-shared';" />

<DoczTabs>
    <Example />
    <Props />
</DoczTabs>
