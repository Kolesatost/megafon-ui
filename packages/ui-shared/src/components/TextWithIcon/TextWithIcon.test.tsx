import * as React from 'react';
import { shallow, mount } from 'enzyme';
import { items } from './doc/TextWithIcon.docz';
import TextWithIcon, { ITextWithIconProps } from './TextWithIcon';

const requiredProps: ITextWithIconProps = {
    items,
};

const allProps: ITextWithIconProps = {
    ...requiredProps,
    title: 'Test title',
    className: 'test-class',
};

describe('<TextWithIcon />', () => {
    it('should render with required props', () => {
        const wrapper = shallow(<TextWithIcon {...requiredProps} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render with all props', () => {
        const wrapper = shallow(<TextWithIcon {...allProps} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should return reference to root element', () => {
        const ref: React.RefObject<HTMLDivElement> = React.createRef();
        mount(
            <TextWithIcon {...allProps} rootRef={ref}>
                <div />
            </TextWithIcon>,
        );

        expect(ref.current).not.toBeNull();
    });
});
