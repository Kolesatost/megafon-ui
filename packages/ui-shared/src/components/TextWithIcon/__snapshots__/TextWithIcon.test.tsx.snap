// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextWithIcon /> should render with all props 1`] = `
<div
  className="mfui-text-with-icon test-class"
>
  <Grid>
    <GridColumn
      desktop="6"
      mobile="12"
      tablet="7"
      wide="6"
    >
      <Header
        as="h5"
        className="mfui-text-with-icon__header"
      >
        Test title
      </Header>
      <div
        className="mfui-text-with-icon__item"
        key="0"
      >
        <div
          className="mfui-text-with-icon__svg-icon"
        >
          <test-file-stub
            style={
              Object {
                "fill": "#00B956",
              }
            }
          />
        </div>
        <div
          className="mfui-text-with-icon__text"
        >
          Привлечение новых клиентов и повышение лояльности действующих
        </div>
      </div>
      <div
        className="mfui-text-with-icon__item"
        key="1"
      >
        <div
          className="mfui-text-with-icon__svg-icon"
        >
          <test-file-stub
            style={
              Object {
                "fill": "#00B956",
              }
            }
          />
        </div>
        <div
          className="mfui-text-with-icon__text"
        >
          Специальные предложения для ваших клиентов
        </div>
      </div>
      <div
        className="mfui-text-with-icon__item"
        key="2"
      >
        <div
          className="mfui-text-with-icon__svg-icon"
        >
          <test-file-stub
            style={
              Object {
                "fill": "#00B956",
              }
            }
          />
        </div>
        <div
          className="mfui-text-with-icon__text"
        >
          Понятная и простая реализация партнерства
        </div>
      </div>
    </GridColumn>
  </Grid>
</div>
`;

exports[`<TextWithIcon /> should render with required props 1`] = `
<div
  className="mfui-text-with-icon"
>
  <Grid>
    <GridColumn
      desktop="6"
      mobile="12"
      tablet="7"
      wide="6"
    >
      <div
        className="mfui-text-with-icon__item"
        key="0"
      >
        <div
          className="mfui-text-with-icon__svg-icon"
        >
          <test-file-stub
            style={
              Object {
                "fill": "#00B956",
              }
            }
          />
        </div>
        <div
          className="mfui-text-with-icon__text"
        >
          Привлечение новых клиентов и повышение лояльности действующих
        </div>
      </div>
      <div
        className="mfui-text-with-icon__item"
        key="1"
      >
        <div
          className="mfui-text-with-icon__svg-icon"
        >
          <test-file-stub
            style={
              Object {
                "fill": "#00B956",
              }
            }
          />
        </div>
        <div
          className="mfui-text-with-icon__text"
        >
          Специальные предложения для ваших клиентов
        </div>
      </div>
      <div
        className="mfui-text-with-icon__item"
        key="2"
      >
        <div
          className="mfui-text-with-icon__svg-icon"
        >
          <test-file-stub
            style={
              Object {
                "fill": "#00B956",
              }
            }
          />
        </div>
        <div
          className="mfui-text-with-icon__text"
        >
          Понятная и простая реализация партнерства
        </div>
      </div>
    </GridColumn>
  </Grid>
</div>
`;
